File: app/src/main/java/com/movie/popcornapp/activities/moviedetails/MovieDetailsViewModel.java
package com.movie.popcornapp.activities.moviedetails;

import androidx.databinding.ObservableField;
import androidx.lifecycle.ViewModel;

import com.movie.popcornapp.models.API.response.MovieResponse;

/**
 * @author george.radu on 2019-07-08.
 */
public class MovieDetailsViewModel extends ViewModel {
    private MovieResponse movieResponse;

    //region Observables
    public ObservableField<String> title = new ObservableField<>();
    public ObservableField<String> year = new ObservableField<>();
    public ObservableField<String> url = new ObservableField<>();
    public ObservableField<String> averageRating = new ObservableField<>();
    public ObservableField<String> voteCount = new ObservableField<>();
    public ObservableField<String> overview = new ObservableField<>();
    //endregion;

    public MovieDetailsViewModel(MovieResponse movieResponse) {
        this.movieResponse = movieResponse;
        setMovieDetails();
    }

    private void setMovieDetails() {
        title.set(movieResponse.getTitle());
        year.set(String.valueOf(movieResponse.getReleaseYear()));
        url.set(movieResponse.getBackdropPath());
        averageRating.set(String.valueOf(movieResponse.getAverageRating()));
        voteCount.set(String.format("%,d", movieResponse.getVoteCount()));
        overview.set(movieResponse.getOverview());
    }
}

File: app/src/main/java/com/movie/popcornapp/activities/moviedetails/MovieDetailsViewModel.kt
package com.movie.popcornapp.activities.moviedetails

import androidx.databinding.ObservableField
import androidx.lifecycle.ViewModel
import com.movie.popcornapp.models.API.response.MovieResponse

class MovieDetailsViewModel(
        private val movieResponse: MovieResponse
) : ViewModel() {

    //region Observables
    var title = ObservableField<String>()
    var year = ObservableField<String>()
    var url = ObservableField<String>()
    var averageRating = ObservableField<String>()
    var voteCount = ObservableField<String>()
    var overview = ObservableField<String>()
    //endregion;

    fun setMovieDetails() {
        title.set(movieResponse.title)
        year.set(movieResponse.releaseYear.toString())
        url.set(movieResponse.backdropPath)
        averageRating.set(movieResponse.averageRating.toString())
        voteCount.set(String.format("%,d", movieResponse.voteCount))
        overview.set(movieResponse.overview)
    }
}